packages/manager/src/features/Images/ImagesLanding.tsx

```diff
@@ -143,11 +141,65 @@ const defaultDialogState = {
   error: undefined,
 };
 
-export const ImagesLanding: React.FC<CombinedProps> = (props) => {
-  useReduxLoad(['images']);
+const preferenceKey = 'images';
 
+export const ImagesLanding: React.FC<CombinedProps> = (props) => {
   const classes = useStyles();
-  const { imagesData, imagesLoading, imagesError, deleteImage } = props;
+
+  const pagination = usePagination(1, preferenceKey);
+
+  const { order, orderBy } = useOrder(
+    {
+      orderBy: 'label',
+      order: 'desc',
+    },
+    `${preferenceKey}-order`
+  );
+
+  const filter = {
+    ['+order_by']: orderBy,
+    ['+order']: order,
+  };
+
+  const {
+    data: images,
+    isLoading: imagesLoading,
+    error: imagesError,
+  } = useImagesQuery(
+    {
+      page: pagination.page,
+      page_size: pagination.pageSize,
+    },
+    filter
+  );
+
+  const { mutateAsync: deleteImage } = useDeleteImageMutation();
+
+  const eventState = useSelector((state: ApplicationState) => state.events);
+  const events = imageEvents(eventState);
+
+  // We want private images and also want the associated events tied in.
+  const imagesItemsById = listToItemsByID(images?.data ?? []);
+  const imagesData = Object.values(imagesItemsById).reduce(
+    (accum, thisImage: Image) =>
+      produce(accum, (draft: any) => {
+        if (!thisImage.is_public) {
+          // NB: the secondary_entity returns only the numeric portion of the image ID so we have to interpolate.
+          const matchingEvent = events.find(
+            (thisEvent) =>
+              `private/${thisEvent.secondary_entity?.id}` === thisImage.id ||
+              (`private/${thisEvent.entity?.id}` === thisImage.id &&
+                thisEvent.status === 'failed')
+          );
+          if (matchingEvent) {
+            draft.push({ ...thisImage, event: matchingEvent });
+          } else {
+            draft.push(thisImage);
```

Comment:
Nice to have: Can we replace `if else` with `ternary` ?  But, it's up to you, totally optional. 
```suggestion
          draft.push(
            !matchingEvent ? thisImage : { ...thisImage, event: matchingEvent }
          );
```